src/Docker/Runner.php

```diff
@@ -334,9 +334,28 @@ public function run(array $jobDefinitions, $action, $mode, $jobId, UsageFileInte
                 ' (row ' . $counter . ' of ' . count($jobDefinitions) . ')'
             );
         }
+        $this->waitForStorageJobs($outputs);
         return $outputs;
     }
 
+    private function waitForStorageJobs(array $outputs)
+    {
+        $jobIds = [];
+        foreach ($outputs as $output) {
+            /** @var Output $output */
+            if ($output->getStorageJobIds()) {
+                $jobIds = array_merge($jobIds, $output->getStorageJobIds());
+            }
+        }
+        $this->loggersService->getLog()->info(sprintf('Waiting for %s storage jobs.', count($jobIds)));
+        foreach ($jobIds as $jobId) {
+            $job = $this->storageClient->waitForJob($jobId);
```

Comment:
tady jsem se trochu sekl, přišlo mi, že by to mohlo pollovat idčka jobů a failovat rychleji, ale tohle je imho ok, pomalejší to nebude a bude to failovat v pořadí, jak se ty joby zakládají (ne dokončují).

Reply:
Mě tohle řešení přijde úplně super. tak jsem si nějak představoval že to bude muset pollovat najednou, proto jsem navrhoval ten multi-get který je ale takhle zbytečný.

Reply:
má to asi jedinou nevýhodu - můžou nastat situace, kdy to vyfailuje zbytečně pozdě.

Reply:
to je celkem jedno ne? jediny, na co to ma vliv je kdy vyfailuje ten syrup job, Vsechny joby do storage uz jsou poslany, takze se stejne uz nemuzou zastavit.